### PR TITLE
Link configure item from property file to java code

### PR DESCRIPTION
**Pull request by chenrujun**

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @saragluna .
Please review. 🙏

**Comment by chenrujun** [comment]
Hi, @saragluna .
Please review and merge. 🙏

**Comment by JimSuplizio** [comment]
There are no engineering systems changes in this PR. Java only changes require reviewers from the Java/Spring. Adding myself, Wes, Mitch or Daniel is only necessary when you're changing versions or things like pom, yml, directory structure etc.

**Comment by chenrujun** [comment]
@JimSuplizio .
I got it. Thank you for your information.

Hi, @saragluna .
Please review and merge. 🙏

**Comment by chenrujun** [comment]
Hi, @JonathanGiles .

I want to use lombok, but I can't use it caused by this line:
https://github.com/Azure/azure-sdk-for-java/blame/master/sdk/parents/azure-client-sdk-parent/pom.xml#L942

Could you please tell me why `annotation processing` is turned off?

**Comment by JonathanGiles** [comment]
> Hi, @JonathanGiles .
> 
> I want to use lombok, but I can't use it caused by this line:
> https://github.com/Azure/azure-sdk-for-java/blame/master/sdk/parents/azure-client-sdk-parent/pom.xml#L942
> 
> Could you please tell me why `annotation processing` is turned off?

I'm a little unclear - this seems to be updating a track one client library code? If so, I'm unclear why you're using the client sdk parent rather than the data sdk parent? Where are you wanting to use lombok? By default we don't use it in our track two libraries.

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProperties.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    public static String getPropertyName(String normalizedName, Property property) {
+        return Stream.of(PREFIX, normalizedName, property.getName())
+            .map(String::trim)
+            .filter(s -> !s.isEmpty())
+            .collect(Collectors.joining(DELIMITER));
+    }
+
+    public static void main(String[] args) {
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProperties.java:172]: Should we remove this `main` method?

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java [hunk 2]
```diff
@@ -56,6 +47,7 @@
 
     KeyVaultEnvironmentPostProcessorHelper(final ConfigurableEnvironment environment) {
         this.environment = environment;
+        Assert.notNull(environment, "environment must not be null!");
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java:50]: Do we really need this assertion here?

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProperties.java (hunk 1 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProperties.java:172]: If I removed `main` method, `mvn clean install` will report `Unused field Error.`

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java (hunk 2 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java:50]: Original code have nonnull check. So I added it, too. Just to keep original logic: https://github.com/Azure/azure-sdk-for-java/pull/12272/files#diff-dcea63cea0abf12a61a1e614cf899b43L204

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProperties.java (hunk 1 above)
Comment (JonathanGiles) [reply sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProperties.java]: It should be removed - we don't want users to run this so I can't see why we would include it.

Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProperties.java]: Already remoted. Thank you for your check. 🙇

**Pull request merged**